helm-org.el

```diff
@@ -58,9 +58,11 @@ NOTE: This will be slow on large org buffers."
   (org-show-entry))
 
 (cl-defun helm-source-org-headings-for-files (filenames
-                                              &optional (min-depth 1) (max-depth 8))
+                                              &optional (min-depth 1) (max-depth 8) (parents nil))
   (helm-build-sync-source "Org Headings"
-    :candidates (helm-org-get-candidates filenames min-depth max-depth)
+    :candidates (helm-org-get-candidates filenames min-depth max-depth parents)
+    :candidate-transformer (if parents 'reverse nil)
```

Comment:
The candidate-transformer is to reverse the list of headings so that they appear in the proper order.  The headline-getting code works by walking up the tree from the current heading, which creates a list with the lowest heading at the top of the list, so it needs to be reversed before being displayed.